### DISCUSSION TITLE
Allow integration with loggers (pino and other) for ability to aggregate all Keystone output in structured mode

### DISCUSSION THREAD

**Discussion by MurzNN** [keystonejs/keystone#7668] [Feature requests]
Now Keystone simply outputs the log lines into stdout via `console.log` that makes it hard to integrate with modern logging libraries like Pino, Winston, Bunyan, npmlog, etc.

To allow the integration we can simply create a middleware library `logger`, that by default just do the same thing - `console.log('string')` but allow replacing it by any custom external library to give flexible log aggregation ability. What do you think about this idea?

**Comment by MurzNN** [comment]
Here are example of logs from Keystone in plaintext, that good to have in JSON format:
```
✨ Starting Keystone
✨ Connecting to the database
✨ Creating server
✅ GraphQL API ready
✨ Preparing Admin UI Next.js app
{"level":30,"time":1657622287549,"pid":28,"hostname":"my-keystone-5978b76984-czvzv","category":"system","operation":"init","msg":"Keystone is initialized."}
✅ Admin UI ready
⭐️ Server Ready on http://localhost:3000
An error occurred handling a request for the Admin UI: TypeError [ERR_INVALID_URL]: Invalid URL: anonymousfox.co
    at new NodeError (internal/errors.js:322:7)
    at onParseError (internal/url.js:270:9)
    at new URL (internal/url.js:346:5)
    at Object.isAccessAllowed (/app/.keystone/admin/.next/server/pages/api/__keystone_api_build.js:9196:94)
    at /app/keystone_core/packages/keystone/dist/createAdminUIMiddleware-ccfdeb37.cjs.dev.js:177:92
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  input: 'anonymousfox.co',
  code: 'ERR_INVALID_URL'
}
```
And only in line 5 there is a custom log record, generated through a 'Pino' logger.

**Comment by rachmanov** [comment]
+1, very useful feature

**Comment by dagual** [comment]
A good idea. The feature will greatly facilitate the support and control of applications.

**Comment by fmquaglia** [comment]
It is a critical feature for running Keystone in major cloud providers.  We're deploying this using Google App Engine, which creates a container inside a managed instance. 
When the container crashes for an unexpected error, we're not able to gather the logs, because user containers' logs are only accessible if using Winston or Bunyan.
